pallets/offchain-demo/src/lib.rs

```diff
@@ -58,17 +58,28 @@ pub mod crypto {
 	use sp_runtime::{
 		app_crypto::{app_crypto, sr25519},
 		MultiSigner, MultiSignature,
+		traits::Verify,
 	};
+	use sp_core::sr25519::Signature as Sr25519Signature;
 
 	app_crypto!(sr25519, KEY_TYPE);
 
 	pub struct TestAuthId;
+	// implemented for ocw-runtime
 	impl frame_system::offchain::AppCrypto<MultiSigner, MultiSignature> for TestAuthId
 	{
 		type RuntimeAppPublic = Public;
 		type GenericSignature = sp_core::sr25519::Signature;
 		type GenericPublic = sp_core::sr25519::Public;
 	}
+
+	// implemented for mock runtime in test
+	impl frame_system::offchain::AppCrypto<<Sr25519Signature as Verify>::Signer, Sr25519Signature> for TestAuthId
+	{
+		type RuntimeAppPublic = Public;
+		type GenericSignature = sp_core::sr25519::Signature;
+		type GenericPublic = sp_core::sr25519::Public;
+	}
 }
 
 // Specifying serde path as `alt_serde`
```

Comment:
There is maybe some ticket system like JIRA or just you are using Issues from Github? I ask because I am curious where/how I find the description of the PR.

Reply:
The description is at the top of this page. This particular PR is a weird one though. It contains a lot of changes from the base branch because this repo (the recipes) is in the process of adopting [gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) to manage PRs